File: python/quickstart.py
"""VecTrade Python SDK — Quick Start Example."""

import os
from vectrade import VecTrade

def main():
    # Initialize client (reads VECTRADE_API_KEY from environment)
    vt = VecTrade()

    # Get a real-time quote
    quote = vt.quotes.get("AAPL")
    print(f"{quote.symbol}: ${quote.price:.2f} ({quote.change_pct:+.2f}%)")

    # Batch quotes
    quotes = vt.quotes.batch(["AAPL", "GOOGL", "MSFT"])
    for q in quotes:
        print(f"  {q.symbol}: ${q.price:.2f}")

    # Stream AI analysis
    print("\n--- AI Analysis ---")
    for chunk in vt.ai.stream("Analyze AAPL for long-term hold"):
        print(chunk.text, end="", flush=True)
    print()


if __name__ == "__main__":
    main()

File: typescript/quickstart.ts
import { VecTrade } from "@vectrade/sdk";

async function main() {
  const vt = new VecTrade();

  // Get a real-time quote
  const quote = await vt.quotes.get("AAPL");
  console.log(`${quote.symbol}: $${quote.price} (${quote.changePct > 0 ? "+" : ""}${quote.changePct}%)`);

  // Batch quotes
  const quotes = await vt.quotes.batch(["AAPL", "GOOGL", "MSFT"]);
  for (const q of quotes) {
    console.log(`  ${q.symbol}: $${q.price}`);
  }

  // Stream AI analysis
  console.log("\n--- AI Analysis ---");
  for await (const chunk of vt.ai.stream("Analyze AAPL for long-term hold")) {
    process.stdout.write(chunk.text);
  }
  console.log();
}

main().catch(console.error);
